code/integration-tests/runtime-tests/tsconfig.json

```diff
@@ -0,0 +1,38 @@
+{
+  "compilerOptions": {
+    // this is specific with augmented overrides
+    "paths": {
+      // this is the package name we use (in the interface imports, --package for generators)
+      "sample-polkadotjs-typegen/*": ["src/*"],
+      // here we replace the @polkadot/api augmentation with our own, generated from chain
+      "@polkadot/api/augment": ["src/interfaces/augment-api.ts"],
+      // replace the augmented types with our own, as generated from definitions
+      "@polkadot/types/augment": ["src/interfaces/augment-types.ts"]
+    },
+    "lib": [
+      "ES2021.String"
+    ],
+    // some other options, whatever you want for your environment
+    "target": "es2017",
+    "module": "commonjs",
```

Comment:
is not that obsolete way of doing modules instread of standard official one?

Reply:
just double check. would prefer we do no use old standards if possible :) neither trying to do FE compability, i more keen on deno/bun/node only with latest TS possible.

Reply:
Changing this caused issues with polkadot api packages, so just to try to integrate this into CI I leave it as it is. But I can look into the issues after CI integration.